### PR TITLE
Add README.md for graphql-compiler package

### PR DESCRIPTION
**Pull request by Yan-J**

### CODE REVIEW

**Comment by wincent** [comment]
This looks great. There are maybe one or two tiny references to Relay in the architecture doc that could be revised, but I don't think it's necessary to block this on that.

File: packages/relay-compiler/graphql-compiler/README.md [hunk 1]
```diff
@@ -1,4 +1,24 @@
...
 
-This is a test release of "graphql-compiler", a code-generation toolkit for GraphQL.
-For now this package is just for testing some internal tooling. 
+GraphQL-Compiler is a code-generation toolkit for GraphQL. It contains the core functionalities of GraphQL code-gen, including file parsing, validation, syntax tree parsing and transformation. The intended audience are developers wanting to use components of GraphQL-Compiler to build their own GraphQL code-gen pipelines.
+
+The GraphQL-Compiler package [exports library code](./GraphQLCompilerPublic.js) which you may use to find the modules you need, or to extend the compiler with your own custom input and output. Note, the internal APIs of the GraphQL-Compiler are under constant iteration, so rolling your own version may lead to incompatibilities with future releases.
+
+The following graph illustrates the high-level architecture of a complete GraphQL code-generation pipeline:
+
+![CodegenPipeline](https://github.com/Yan-J/relay/raw/master/packages/relay-compiler/graphql-compiler/docs/Architecture.png)
```
Comment (edvinerikson) [inline packages/relay-compiler/graphql-compiler/README.md]: shouldn't this point to facebook/relay?

Comment (wincent) [inline packages/relay-compiler/graphql-compiler/README.md]: Aye. Good catch @edvinerikson.

**Comment by leebyron** [comment]
@Yan-J could you please re-sign the CLA with your github username? https://code.facebook.com/cla/individual

**Comment by alloy** [comment]
@leebyron @jstejada The CLA has been signed.

**Pull request closed without merging**